client/dashboard/sites/settings-php/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { DataForm } from '@automattic/dataviews';
+import { useQuery, useMutation } from '@tanstack/react-query';
+import {
+	Card,
+	CardBody,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { getPHPVersions } from 'calypso/data/php-versions';
+import { siteQuery, sitePHPVersionQuery, sitePHPVersionMutation } from '../../app/queries';
+import PageLayout from '../../components/page-layout';
+import SettingsPageHeader from '../settings-page-header';
+import { canUpdatePHPVersion } from './utils';
+import type { Field } from '@automattic/dataviews';
+
+export default function PHPVersionSettings( { siteSlug }: { siteSlug: string } ) {
+	const { data: site } = useQuery( siteQuery( siteSlug ) );
+	const canUpdate = site && canUpdatePHPVersion( site );
+
+	const { data: version } = useQuery( {
+		...sitePHPVersionQuery( siteSlug ),
+		enabled: canUpdate,
+	} );
+	const mutation = useMutation( sitePHPVersionMutation( siteSlug ) );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const [ formData, setFormData ] = useState< { version: string } >( {
+		version: version ?? '',
+	} );
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const { phpVersions } = getPHPVersions();
+
+	const fields: Field< { version: string } >[] = [
+		{
+			id: 'version',
+			label: __( 'PHP version' ),
+			Edit: 'select',
+			elements: phpVersions.filter( ( option ) => {
+				// Show disabled PHP version only if the site is still using it.
+				if ( option.disabled && option.value !== version ) {
+					return false;
+				}
+				return true;
+			} ),
+		},
+	];
+
+	const form = {
+		type: 'regular' as const,
+		fields: [ 'version' ],
+	};
+
+	const isDirty = formData.version !== version;
```

Comment:
Nit: Would it make sense to name `version` to `currentVersion` for better readability, and in case formData.version is deconstructed?

Reply:
Good idea, done!